### PR TITLE
Avoided creating default form fields in fields_for_model() when declared on form.

### PR DESCRIPTION
**Pull request by felixxm**
Noticed when reviewing #16614.

### CODE REVIEW

File: django/forms/models.py [hunk 1]
```diff
@@ -177,6 +178,7 @@ def fields_for_model(
     ``apply_limit_choices_to`` is a boolean indicating if limit_choices_to
     should be applied to a field's queryset.
```
Comment (nessita) [inline django/forms/models.py:179]: Seems like docs for this new param would be needed?

File: django/forms/models.py [hunk 2]
```diff
@@ -319,8 +325,7 @@ def __new__(mcs, name, bases, attrs):
                 message = "Unknown field(s) (%s) specified for %s"
                 message %= (", ".join(missing_fields), opts.model.__name__)
                 raise FieldError(message)
-            # Override default model fields with any custom declared ones
-            # (plus, include all the other declared fields).
+            # Include all the other declared fields.
             fields.update(new_class.declared_fields)
```
Comment (nessita) [inline django/forms/models.py:329]: Do you think there is any way to call this `update` with the remaining of `new_class.declared_fields` and not pass the whole thing which was already (almost) consumed?

**Review by nessita — COMMENTED** [review]
Looks good! I was wondering if a small test could be added to avoid regressions. What do you think?

File: django/forms/models.py (hunk 2 above)
Comment (felixxm) [reply django/forms/models.py:329]: For this, we'd have to create a new dictionary. Probably not worth doing :thinking: .

**Comment by felixxm** [comment]
@nessita Thanks for checking :+1: pushed edits.

**Pull request merged**